### PR TITLE
Feature add Moonbeam's parachain staking precompile

### PR DESCRIPTION
**Pull request by magecnion**
Previously merged #673 included the minimal changes necessary to accommodate the code from Moonbeam's precompile staking module so we can leverage its implementation:
- Increasing the cost of three tested calls (this is reflected within the test file).
- Removing some associated types from the mock file.
- Removing the 4th parameter of the `try_dispatch` function, as it doesn't exist in our version.
- Adding `rewards_account = None` in the mock.

**The incremental changes included in the current PR are only the e2e tests.**

### CODE REVIEW

**Review by asiniscalchi — DISMISSED** [review]

File: precompiles/parachain-staking/StakingInterface.sol [hunk 1]
```diff
@@ -0,0 +1,305 @@
+// SPDX-License-Identifier: GPL-3.0-only
```
Comment (luispdm) [inline precompiles/parachain-staking/StakingInterface.sol:1]: We have all of the other contracts without `Interface` in their filenames. Can we rename this file `Staking.sol`, please?

File: e2e-tests/tests/util.ts [hunk 2]
```diff
@@ -41,17 +44,33 @@ export async function customRequest(web3: Web3, method: string, params: any[]) {
...
+			web3: null,
+			ethersjs: null,
+			polkadot: undefined,
+		};
+
+		before(async () => {
+			if (providerNodeUrl) {
+				context.web3 = new Web3(providerNodeUrl);
+				const wsProvider = new HttpProvider(providerNodeUrl);
+				context.polkadot = await new ApiPromise({ provider: wsProvider }).isReady;
```
Comment (luispdm) [inline e2e-tests/tests/util.ts:67]: You can remove this line and move line 71 after the `else` block

**Review by luispdm — CHANGES_REQUESTED** [review]

File: e2e-tests/tests/util.ts (hunk 2 above)
Comment (magecnion) [reply e2e-tests/tests/util.ts:67]: In that case, I would need to create an empty `let wsProvider` variable before the `if` statement, which will then be assigned a value depending on the condition executed. I think it's clearer to have the code executed together, with the minor drawback of having a duplicated line.

File: precompiles/parachain-staking/StakingInterface.sol (hunk 1 above)
Comment (magecnion) [reply precompiles/parachain-staking/StakingInterface.sol:1]: Sure, I will use `ParachainStaking` (which is the name of the pallet)

**Review by asiniscalchi — APPROVED** [review]

**Pull request merged**